streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBTimeOrderedKeyValueBytesStore.java

```diff
@@ -42,7 +46,29 @@ protected KeyValue<Bytes, byte[]> getIndexKeyValue(final Bytes baseKey, final by
 
     @Override
     Map<KeyValueSegment, WriteBatch> getWriteBatches(final Collection<ConsumerRecord<byte[], byte[]>> records) {
-        throw new UnsupportedOperationException("Do not use for TimeOrderedKeyValueStore");
+        final Map<KeyValueSegment, WriteBatch> writeBatchMap = new HashMap<>();
```

Comment:
Looks like this PR is now also adding restoration logic, in addition to comparing table history retention and join grace period. Was that intentional? Whenever we add the restoration logic, we'll want to add integration tests for that -- maybe it'd be easier to have it be a separate PR. WDYT?

Reply:
Yeah that should be in a different PR, not sure how that snuck in